kolibri/plugins/learn/assets/src/views/content-card-carousel/index.vue

```diff
@@ -0,0 +1,255 @@
+        return {
+          width: `${this.contentSetSize * contentCardWidth}px`,
+        };
+      },
+    },
+    methods: {
+      beforeEnterStyle(el) {
+        const sign = this.animation === 'next' ? '' : '-';
+        el.style.transform = `translateX(${sign}${this.contentSetSize * contentCardWidth}px)`;
+        el.style.opacity = 0;
+      },
+      enterStyle(el, done) {
+        window.setTimeout(() => {
+          el.style.opacity = '';
+          el.style.transform = '';
+          done();
+        }, 500);
+      },
+      leaveStyle(el, done) {
+        const sign = this.animation === 'next' ? '-' : '';
+        el.style.transform = `translateX(${sign}${this.contentSetSize * contentCardWidth}px)`;
+        el.style.opacity = 0;
+
+        window.setTimeout(done, 500);
+      },
+      nextSet() {
+        const lastIndex = this.contents.length - 1;
+
+        this.controlCounter += 1;
+
+        const nextEndIndex = (this.nextContentSetStartIndex + this.contentSetSize) - 1;
+        if (nextEndIndex > lastIndex) {
+          this.contentSetStartIndex = this.contents.length - this.contentSetSize;
+        } else {
+          this.contentSetStartIndex = this.nextContentSetStartIndex;
+        }
+
+        if (this.isLastSet) {
+          this.$emit('end');
+        }
+
+        this.animation = 'next';
+      },
+      previousSet() {
+        this.controlCounter += 1;
+
+        const prevStartIndex = this.contentSetStartIndex - this.contentSetSize;
+        if (prevStartIndex < 0) {
+          this.contentSetStartIndex = 0;
+        } else {
+          this.contentSetStartIndex = prevStartIndex;
+        }
+
+        this.animation = 'previous';
+      },
+    },
+  };
+
+</script>
+
+
+<style lang="stylus" scoped>
+
+  @require '~kolibri.styles.definitions'
+
+  $card-height = 210px
+
+  .content-carousel
+    margin-top: 1em
+    margin-bottom: 1em
+    clearfix()
+
+    &-details
+      clearfix()
+      &-header
+        float: left
+        text-align: left
+        margin-bottom: 1em
+        h2
+          margin: 0
+      &-view-all
+        float: right
+        color: white
+        background-color: $core-action-normal
+
+    &-set
+      margin-left: auto
+      margin-right: auto
+      overflow: hide
+
+    &-controls
+      // set up the parent element that the buttons use for reference
+      position: absolute
+      width: 100%
+      clearfix()
+      .next, .previous
+        // uses parent div as reference
+        position: absolute
+        top: ($card-height / 2)
+
+        // using material definition for resting Raised Button
+        z-index: 2
+        box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
+        &:active
+          z-index: 8
+          box-shadow: 0 10px 20px rgba(0,0,0,0.19), 0 6px 6px rgba(0,0,0,0.23);
+      .next
+        right: 0
+      .previous
+        left: 0
+
+  .content-card
+    transition: all 0.5s ease
```

Comment:
I would try to avoid applying transitions to `all` when reasonably possible. It might have unintended side effects - both visual and performance. Also makes the developer's intentions less clear - better to be explicit about what property is intended to be transitioned.